### ISSUE TITLE
`reportUnnecessaryIsInstance` false positive with `strictGenericNarrowing`

### ISSUE THREAD

**Issue by BarrensZeppelin** [DetachHead/basedpyright#905]
```py
from collections.abc import Callable


type MyCallable = Callable[[], None]

@staticmethod
def f() -> None:
    pass


def g(f: MyCallable):
    if isinstance(f, staticmethod): # Unnecessary isinstance call; "MyCallable" is never an instance of "staticmethod[..., object]"  (reportUnnecessaryIsInstance)
        print("foo")
        
g(f)
```
[Playground](http://basedpyright.com/?typeCheckingMode=standard&reportUnnecessaryIsInstance=true&strictGenericNarrowing=true&code=GYJw9gtgBAxmA28CmMAuBLMA7AzgOgEMAjGKdCABzBFSgGEDFjkAoN1ATwqSgFkOGTIsigBeeo3jMkAbRkBdADRQActiTy2AARyoCGGBCSoAFmAAmLc0mBRgACgCUUALQA%2BVeoBcLKH6gUBDg4bFY2UADm9sBefAKS0o4%2B-mS26DjouHpYMEjRyrr66IbGZuZJvin%2BFCCZqPYARMBgYA2OlVVQLFHAjkA)

The type checker thinks that a value that's assignable to `type MyCallable = Callable[[], None]` can never be an instance of `staticmethod`, but this is incorrect.

**Comment by DetachHead** [comment]
i guess this is caused by the fact that `Callable` is a complete dumpsterfire thats riddled with special-casing... recreating this example without it does not reproduce the issue:

```py
from typing import Protocol

class MyCallable[**P, T](Protocol):
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> T: ...


def g(f: MyCallable[[], None]):
    if isinstance(f, staticmethod):
        reveal_type(f)
```
